testament: fix missing error reporting

zerbina opened the pull request:
## Summary

Fix tests with invalid specs being silently ignored (they were neither
compiled nor run) when using `all` or `cat megatest`.

## Details

The `isJoinableSpec` procedure didn't prevent tests with an invalid
spec (`reInvalidSpec`) from being joined, which overrode the failure
with `reJoined`. Since the specs then had no target enabled, they were
also not included in the megatest.

A few invalid test specifications went by unnoticed due to this bug --
they're now fixed.

zerbina committed:
testament: don't join tests with invalid specs

If a test's specification is invalid, it must not be joined, so that an
error can be reported for it.

Previously, a test with an invalid spec could be joined, which
effectively discarded the error. The test was also neither run nor
compiled, since no targets were enabled for the test, in that case.

zerbina committed:
tests: fix the bogus specs

saem commented:
/merge

saem commented on tests/lang_callable/macros/tmodify_result_assignment.nim:
```diff
@@ -3,7 +3,7 @@ discard """
   '''
-  errmsg: "got <string> but expected 'int'"
+  errormsg: "got <string> but expected 'int'"
   line: 26
```
```suggestion
  line: 25
```

This will appease the test, but I'm not sure if that's right given the `copyLineInfo` on line 22.

zerbina replied:
The error should be on line 26, but the `copyLineInfo` copies to the wrong target node (`result.body` instead of `result.body[0][1]`).

zerbina committed:
tests: fix `tmodify_result_assignment.nim`

The line info of the assignment source expression didn't match the one
expected by the test.

zerbina committed:
tests: fix `tfor_ill_formed_sym.nim`

saem commented:
/merge
